airbyte-db/lib/src/main/java/io/airbyte/db/instance/jobs/migrations/V0_29_15_001__Add_temporalWorkflowId_col_to_Attempts.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.db.instance.jobs.migrations;
+
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.jooq.DSLContext;
+import org.jooq.impl.DSL;
+import org.jooq.impl.SQLDataType;
+
+public class V0_29_15_001__Add_temporalWorkflowId_col_to_Attempts extends BaseJavaMigration {
+
+  @Override
+  public void migrate(Context context) throws Exception {
+    // Warning: please do not use any jOOQ generated code to write a migration.
+    // As database schema changes, the generated jOOQ code can be deprecated. So
+    // old migration may not compile if there is any generated code.
+    DSLContext ctx = DSL.using(context.getConnection());
+    ctx.alterTable("attempts").addColumn(DSL.field("temporal_workflow_id", SQLDataType.VARCHAR.nullable(false).defaultValue("")))
```

Comment:
> `SQLDataType.VARCHAR.nullable(false)`

This is not backward compatible. Ideally we should add nullable columns first.

Reply:
If we decide to use `VARCHAR`, do we know how long this string will be? Better set a limit for it if we have a rough idea about the format of the ID.

Reply:
No real reason. I thought a default value is clearer. Nullable field is fine for me. Good point on the backward compatibility. Will switch to nullable.

> If we decide to use VARCHAR, do we know how long this string will be? Better set a limit for it if we have a rough idea about the format of the ID.

Not clear. I looked at the documentation and there are no explicit limits. The return type is string, which is why I've left it as a flexible varchar.

Reply:
If this field is UUID strings, and you don't want to declare it as UUID type, you can do `SQLDataType.VARCHAR(36)`.

Reply:
[Looks](https://stackoverflow.com/a/8295254) like there isn't real benefit to specifying a length beyond enforcing a constraint check. Since Temporal doesn't explicitly state this is a UUID, I'm leaning towards leaving this as a flexible varchar.

Reply:
I didn't realise this was being translated to an insane limit under the hood. I'm going to use 256 for now to give us some buffer.